src/components/results/use-results-tab.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { ShortcircuitAnalysisResultTabs } from './shortcircuit/shortcircuit-analysis-result.type';
+
+export enum ResultsTabsLevel {
+    ROOT = 0,
+    ONE = 1,
+}
+
+export enum ResultsTabsRootLevel {
+    LOADFLOW = 0,
+    SECURITY_ANALYSIS = 1,
+    SENSITIVITY_ANALYSIS = 2,
+    SHORTCIRCUIT_ANALYSIS = 3,
+    DYNAMIC_SIMULATION = 4,
+    VOLTAGE_INIT = 5,
+}
+
+// to fill with other first level tabs when needed (ex : ShortcircuitAnalysisResultTabs | SensitivityAnalysisResultTabs | ...)
+type ResultsTabsLevelOne = ShortcircuitAnalysisResultTabs;
+
+export type ResultTabIndexRedirection =
+    | [ResultsTabsRootLevel, ResultsTabsLevelOne]
+    | null;
+
+/**
+ * handles redirection to specific tab
+ * @param resultTabIndexRedirection array holding the desired tabs to be redirected to [2, 1] would redirect to tab number 2, subtab number 1
+ * @param setTabIndex setter allowing to set the tab in the current component
+ * @param tabLevel tab level of the current component
+ */
+export const useResultsTab = (
+    resultTabIndexRedirection: ResultTabIndexRedirection,
+    setTabIndex: (index: number) => void,
+    tabLevel: ResultsTabsLevel
+) => {
+    useEffect(() => {
```

Comment:
antipattern of useeffect https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

also kind of an antipattern I feel to "hide" this basically one line in a custom hook, maybed you sensed that something was wrong..?

Anyway all this should be removed I think, and keep only one state for the tab index

Reply:
To match the "showInSpreadsheet" behaviour, we will keep this behaviour for now, with TODO comments in order to fix the tab system in the future